source/common/router/router.cc

```diff
@@ -305,8 +305,22 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
 }
 
 Http::ConnectionPool::Instance* Filter::getConnPool() {
+  // Choose protocol based on cluster configuration and downstream connection
+  // Note: Cluster may downgrade HTTP2 to HTTP1 based on runtime configuration.
+  auto features = cluster_->features();
+
+  ENVOY_STREAM_LOG(debug, "getConnPool cluster features: {:x}", *callbacks_, features);
+
+  const Optional<Http::Protocol>& downstream_protocol = callbacks_->requestInfo().protocol();
+  enum Http::Protocol protocol =
```

Comment:
nit: This multiple tertiary operator is a bit of an eyeful. Can you break apart?

Reply:
Refactored.